crypto/certificate.go

```diff
@@ -0,0 +1,114 @@
+package crypto
+
+import (
+	"errors"
+)
+
+// Signable defines an interface which would generate a unique hash based on its content
+type Signable interface {
+	Hash() []byte
+}
+
+// PublicKey defines an interface of Public Key
+type PublicKey interface {
+	Bytes() []byte
+	Str() string
+}
+
+// Signature defines an interface of
+type Signature interface {
+	Bytes() []byte
+}
+
+// Verifier defines an interface which could verify a signature
+type Verifier interface {
+	Verify(signature Signature) bool
```

Comment:
Why not Verify(sig, pk)?

Reply:
I assumed that the interface defined here contains the public key. What you suggested above is more like the interface of crypto algorithm